server/service/transport.go

```diff
@@ -74,7 +75,11 @@ func nameFromRequest(r *http.Request, varName string) (string, error) {
 	if !ok {
 		return "", errBadRoute
 	}
-	return name, nil
+	unescaped, err := url.PathUnescape(name)
```

Comment:
I verified this function is only called from the routes we intend to change, so this should not have unintended effects on other routes.